src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+            default=None,
+            domain=IsInstance((str, list)),
+            description="Registers species to open to optimization, this can help with solvability of some problems",
+            doc="""Registers species (or list of species) to open to optimization and write empty constraint for,
+            this can help with solvability of some problems, but can
+            lead to unexpected results depending on database, activity coefficients, and inputs chosen.
+            This generally should not be left as None""",
+        ),
+    )
+    CONFIG.declare(
+        "open_species_on_speciation_block",
+        ConfigValue(
+            default=None,
+            domain=IsInstance((str, list)),
+            description="Registers species to open to optimization, this can help with solvability of some problems",
+            doc="""Registers species (or list of species) to open to optimization and write empty constraint for,
+            this can help with solvability of some problems, but can
+            lead to unexpected results depending on database, activity coefficients, and inputs chosen.
+            This generally should not be needed""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+        """ configure state"""
+        if self.config.build_speciation_block:
+            """create spectitation block and then property block"""
+            self.speciation_block = Block()
+            self.build_rkt_state(self.speciation_block, speciation_block=True)
+            self.build_rkt_inputs(self.speciation_block, speciation_block=True)
+            self.build_rkt_outputs(self.speciation_block, speciation_block=True)
+            self.build_rkt_jacobian(self.speciation_block)
+            self.build_rkt_solver(self.speciation_block, speciation_block=True)
+            self.build_gray_box(self.speciation_block)
+
+            self.build_rkt_state(
+                self,
+                speciation_block=False,
+                speciation_block_built=True,
+                input_composition=self.speciation_block.outputs,
+            )
+            self.build_rkt_inputs(
+                self, speciation_block=False, speciation_block_built=True
+            )
+            self.build_rkt_outputs(self, speciation_block=False)
+            self.build_rkt_jacobian(self)
+            self.build_rkt_solver(self, speciation_block=False)
+            self.build_gray_box(self)
+        else:
+            """create property block only"""
+            self.build_rkt_state(self)
+            self.build_rkt_inputs(self)
+            self.build_rkt_outputs(self)
+            self.build_rkt_jacobian(self)
+            self.build_rkt_solver(self)
+            self.build_gray_box(self)
+
+    def build_rkt_state(
+        self,
+        block,
+        speciation_block=False,
+        speciation_block_built=False,
+        input_composition=None,
+    ):
+        """this will buld our rkt state specified block.
```

Comment:
Typo: Capital to start doc string, "buld" -> "build".

Should "rkt" be expanded for later readers of the code?